### ISSUE TITLE
java_tools release broken

### ISSUE THREAD

**Issue by meisterT** [bazelbuild/bazel#10023]
### Description of the problem:

java_tools release probably broke a couple of Java projects (with Dagger?)

See for a full list of breakages in downstream:
https://buildkite.com/bazel/bazel-at-head-plus-downstream/builds/1234#a822417e-f2f7-4927-abe7-b01289011f67
(Note that it includes more problems than this particular issue.)

One example:
https://buildkite.com/bazel/bazel-at-head-plus-downstream/builds/1234#83e0dae2-b545-413c-9311-56b79320a2b6

Example error:
```
ERROR: /var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/ecf942666dab041fc60b1286a219db3e/external/io_bazel_rules_kotlin/src/main/kotlin/io/bazel/kotlin/builder/BUILD:34:1: Couldn't build file external/io_bazel_rules_kotlin/src/main/kotlin/io/bazel/kotlin/builder/libbuilder.jar: Building external/io_bazel_rules_kotlin/src/main/kotlin/io/bazel/kotlin/builder/libbuilder.jar (2 source files) and running annotation processors (ComponentProcessor) failed (Exit 1)
--
  | bazel-out/host/bin/external/io_bazel_rules_kotlin/src/main/kotlin/io/bazel/kotlin/builder/_javac/builder/libbuilder_sourcegenfiles/io/bazel/kotlin/builder/KotlinBuilderComponent_Module_ProvidePluginArgEncoderFactory.java:23: error: [RefersToDaggerCodegen] Don't refer to Dagger's internal or generated code
  | return providePluginArgEncoder(module, toolchainProvider.get());
  | ^
  | (see https://errorprone.info/bugpattern/RefersToDaggerCodegen)
  | bazel-out/host/bin/external/io_bazel_rules_kotlin/src/main/kotlin/io/bazel/kotlin/builder/_javac/builder/libbuilder_sourcegenfiles/io/bazel/kotlin/builder/KotlinBuilderComponent_Module_ProvidePluginArgEncoderFactory.java:33: error: [RefersToDaggerCodegen] Don't refer to Dagger's internal or generated code
  | return Preconditions.checkNotNull(instance.providePluginArgEncoder(toolchain), "Cannot return null from a non-@Nullable @Provides method");
  | ^
  | (see https://errorprone.info/bugpattern/RefersToDaggerCodegen)
```

### What's the output of `bazel info release`?

This is with the 1.1.0rc1

**Comment by meisterT** [comment]
cc @hlopko who created the release

**Comment by iirina** [comment]
It looks related to the error prone upgrade in https://github.com/bazelbuild/bazel/pull/9372, I'm rebuilding the tools and update this thread when I confirm the culprit.

**Comment by iirina** [comment]
@meisterT I can confirm the error prone upgrade to 2.3.3 is the culprit. The new error prone release introduced new error prone checks. This should have been an incompatible change. @cushon FYI

Next steps:
1. revert d990746fbd74334aee195dbdb2f6658f30108113
2. create java_tools patch release v6.1
3. upgrade the default java_tools in bazel to v6.1
4. cherry-pick commit for step 3 into bazel 1.1

**Comment by iirina** [comment]
This should be fixed by 0a8071af47223989e9ba335e63b8a9595fe6f10f.
